### PR TITLE
Images memory usage optimisation

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**
Use custom solution for in-memory cache instead of NSCache. 
ImageLoadingService refactoring. 
Cache downsampled images with different settings under different keys separately.
Apply downsampling with .icon, .mid, .max sizes where appropriate.
Added UnitTests for ImageLoadingService and ImageCacheStorage

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+// MARK: - Open methods
+extension ImagesCacheStorage: ImagesCacheStorageProtocol {
+    var cacheMemoryUsage: Int {
+        imageCache.values.map({ $0.memoryUsage }).reduce(0, { $0 + $1 })
+    }
+    var numberOfCachedItems: Int { imageCache.count }
+    
+    func cachedImage(for key: String) -> UIImage? {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift:34]: Verb 'get'?

File: unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                        break
+                    }
+                }
+            }
+            
+            self.imageCache[.init(key)] = image
+            
+            
+#if DEBUG
+            print("Did cache image with size \(image.size) for key \(key)\nCurrent images cache memory usage: \(cacheMemoryUsage)")
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift:69]: printInfo()

**Review by rommex — APPROVED** [review]
with minor suggestions

File: unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift (hunk 1 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift:34]: Updated

File: unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift (hunk 2 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService/ImagesCacheStorage.swift:69]: Fixed

**Pull request merged**